### PR TITLE
Improve Extensibility of VideoPress Privacy Setting filter by adding the embedded post id.

### PR DESCRIPTION
**Pull request by pgk**
Enhances the privacy setting filter by adding the id of the post that embeds the video as an argument. This is useful because we might want to tie authorization mechanisms that require a post_id to work. Examples are 

* https://woocommerce.com/products/woocommerce-memberships/
* https://woocommerce.com/products/woocommerce-subscriptions/
* https://wordpress.org/plugins/sensei-lms/

Fixes #

#### Changes proposed in this Pull Request:

* Adds an `$embedded_post_id` parameter to the `videopress_is_current_user_authed_for_video` filter.
* Delays enqueueing of token bridge script a bit so we have a post id.

#### Testing instructions:

Here is a small mu-plugin to test with...

```php
<?php
function example_override_video_privacy_setting_for_post( $is_user_authed, $guid, $embedded_post_id = 0 ) {
	$your_post_id = null; /* TODO: add your post id here. */
	if ( $your_post_id === $embedded_post_id ) {
		return true;
	}

	return $is_user_authed;
}

function videopress_override_init() {
	add_filter( 'videopress_is_current_user_authed_for_video', 'example_override_video_privacy_setting_for_post', 10, 3 );
}
add_action( 'init', 'videopress_override_init' );

```

* this can be tested on a local dev env, or in any env where you can add an mu-plugin or a bit of php code.
* Set global privacy setting to private.
* Create and publish 2 posts and embed the same video in both. Note post_ids
* Add the php snippet above to your site. Replace the check with your post id.
* Test the post pages on a private window. The video is public on the post with the id you used on the snippet.

### CODE REVIEW

File: projects/plugins/jetpack/modules/videopress/class.videopress-ajax.php [hunk 1]
```diff
@@ -72,13 +72,19 @@ private function is_valid_guid( $guid ) {
 	 * @return void
 	 */
 	public function wp_ajax_videopress_get_playback_jwt() {
-		$guid = filter_input( INPUT_POST, 'guid' );
+		$guid             = filter_input( INPUT_POST, 'guid' );
+		$embedded_post_id = filter_input( INPUT_POST, 'post_id', FILTER_VALIDATE_INT );
+
+		if ( false === $embedded_post_id ) {
```
Comment (thedebian) [inline projects/plugins/jetpack/modules/videopress/class.videopress-ajax.php:78]: filter_input may also return null, you could either check empty( $embedded_post_id ) or use the options array with default => 0 like

`$embedded_post_id = filter_input( INPUT_POST, 'post_id', FILTER_VALIDATE_INT, array( 'options' => array( 'default' => 0 ) );`

Comment (thedebian) [inline projects/plugins/jetpack/modules/videopress/class.videopress-ajax.php:78]: filter_input may also return null, you could either check empty( $embedded_post_id ) or use the options array with default => 0 like

`$embedded_post_id = (int) filter_input( INPUT_POST, 'post_id', FILTER_VALIDATE_INT, array( 'options' => array( 'default' => 0 ) );`

**Review by thedebian — CHANGES_REQUESTED** [review]

File: projects/plugins/jetpack/modules/videopress/class.videopress-ajax.php [hunk 2]
```diff
@@ -72,13 +72,20 @@ private function is_valid_guid( $guid ) {
...
-		$guid = filter_input( INPUT_POST, 'guid' );
-		if ( ! $this->is_valid_guid( $guid ) ) {
+		$guid             = filter_input( INPUT_POST, 'guid' );
+		$embedded_post_id = filter_input( INPUT_POST, 'post_id', FILTER_VALIDATE_INT );
+
+		if ( empty( $embedded_post_id ) ) {
+			$embedded_post_id = 0;
+		}
+
+		if ( empty( $guid ) && ! $this->is_valid_guid( $guid ) ) {
```
Comment (thedebian) [inline projects/plugins/jetpack/modules/videopress/class.videopress-ajax.php:82]:
```suggestion
		if ( empty( $guid ) || ! $this->is_valid_guid( $guid ) ) {
```

**Review by thedebian — CHANGES_REQUESTED** [review]
Another slight change (I think), apart from that, code is 👍, I still need to test it, but it'll be done ASAP

**Comment by pablohoneyhoney** [comment]
Neat!

[Left a few notes in P2](https://videop2.wordpress.com/2022/04/20/introducing-the-jetpack-video-privacy-setting/).

- [ ] Can we please take a look at the private message to be more intentional in its design @evilluendas?

- [ ] The text below the privacy selector could be dynamic and contextual to the selected privacy setting. The default state shall ideally be instructional of the interaction and also tell you what the site setting is currently.

- [ ] And the tab title probably would be clearer if it's Video Privacy, or something in those lines. </p>

- [ ] There's also a strange spinner when loading the video in the editor that could be updated to [the latest wp spinner](https://wordpress.github.io/gutenberg/?path=/story/components-spinner--default).

**Review by thedebian — DISMISSED** [review]
Worked well ! 👍 :shipit:

**Review by thedebian — APPROVED** [review]
Re-approved !

**Comment by pgk** [comment]
r245781-wpcom

**Pull request merged**